### DISCUSSION TITLE
tensorflow

### DISCUSSION THREAD

**Discussion by TheCodeCache** [flet-dev/flet#3246] [Packages] [answered]
https://pypi.org/project/tensorflow/

**Comment by ndonkoHenri** [comment, marked as answer]
Unfortunately, full TensorFlow currently can't be packaged for a Flet mobile app:

- No source distribution on PyPi: Every TensorFlow release ships only pre-built platform wheels (Linux/macOS/Windows) — there's no sdist, so there's nothing for Flet's mobile build to cross-compile from.
- Bazel-only build: TensorFlow is built exclusively with Bazel, not setuptools/CMake, so the iOS/Android cross-compile toolchain Flet uses can't drive its build.
- Size: The compiled native runtime (libtensorflow + XLA + MLIR + Eigen + abseil) is ~220–570 MB per platform — far beyond what's reasonable to embed in an app?

Possible alternative: run inference on a backend service and call it from the app.
